docs/book/src/jobs.md

```diff
@@ -0,0 +1,42 @@
+# Jobs
+
+This document intents to provide an overview over our jobs running via Prow and Github actions.
+
+## Build and Tests running on the default branch
+
+
+ > NOTE : To see which test jobs execute which e2e tests, you can click on the links which lead to the
+  respective test overviews in testgrid.
+
+### Presubmits
+
+  Prow Presubmits:
+
+  - [pull-cluster-api-provider-azure-build] `./scripts/ci-build.sh` REQUIRED
+  - [pull-cluster-api-provider-azure-test] `./scripts/ci-test.sh`   REQUIRED
+  - [pull-cluster-api-provider-azure-e2e] `./scripts/ci-e2e.sh`     REQUIRED
+  - [pull-cluster-api-provider-azure-e2e-exp] `./scripts/ci-e2e.sh`
+  - [pull-cluster-api-provider-azure-windows] `./scripts/ci-e2e.sh` REQUIRED
+  - [pull-cluster-api-provider-azure-verify]                        REQUIRED
```

Comment:
pull-cluster-api-provider-azure-verify: runs `make verify`. It performs several validations including go modules, file headers, generated files, and shellcheck